### DISCUSSION TITLE
Can HammerDB connect to an existing oracle database and use the user's own actual business SQL for stress testing without creating example data?

### DISCUSSION THREAD

**Discussion by jieguolove** [TPC-Council/HammerDB#693] [Q&A] [answered]
Can HammerDB connect to an existing oracle database and use the user's own actual business SQL for stress testing without creating example data? If it can, please give an example of configuration, thank you!

**Comment by sm-shaw** [comment, marked as answer]
Yes you can replay your own Oracle workload using the oracle trace file replay feature https://www.hammerdb.com/docs/ch14.html
There is a very good blog on this from house of brick https://houseofbrick.com/blog/hammerdb-capture-and-replay-for-oracle/ that you can use as a guide.

**Reply by jieguolove** [reply]
It helps me a lot.Thanks a lot!

**Comment by jieguolove** [comment]
why the error?
Script editor:
`#!/usr/local/bin/tclsh8.6
package require Oratcl
####UPDATE THE CONNECT STRING BELOW###
set connect jyc/jyc@192.168.207.164:1521/jyc
set lda [oralogon $connect]
set curn140649325792368 [oraopen $lda ]
set sql140649325792368 "SELECT DECODE(USER, 'XS\$NULL', XS_SYS_CONTEXT('XS\$SESSION','USERNAME'), USER) FROM SYS.DUAL "
orasql $curn140649325792368 $sql140649325792368
set row [orafetch $curn140649325792368 -datavariable output ]
while { [ oramsg  $curn140649325792368 ] == 0 } {
puts $output
set row [orafetch  $curn140649325792368 -datavariable output ]
}
oraclose $curn140649325792368
set curn140649325792368 [oraopen $lda ]
set sql140649325792368 "BEGIN DBMS_APPLICATION_INFO.SET_MODULE(:1,NULL); END; "
oraparse $curn140649325792368 $sql140649325792368
oraplexec $curn140649325792368 $sql140649325792368 :1 {SQL*Plus} 
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select count(*) from t1 "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select table_name from user_tables "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select * from t2 where rownum<2 "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select count(*) from t2 where username='JYC' "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select count(*) from t3 "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select count(*) from t5 where username='SYS' "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "select count(*) from t5 where username='SYSTEM' "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
set curn140649325788728 [oraopen $lda ]
set sql140649325788728 "SELECT count(*) from t5 "
orasql $curn140649325788728 $sql140649325788728
set row [orafetch $curn140649325788728 -datavariable output ]
while { [ oramsg  $curn140649325788728 ] == 0 } {
puts $output
set row [orafetch  $curn140649325788728 -datavariable output ]
}
oraclose $curn140649325788728
oralogoff $lda
`
![image](https://github.com/TPC-Council/HammerDB/assets/96612712/cc8ce1de-20b7-43cb-9f52-498cc2ddf5e2)
Transaction Counter Error: failed to load common transaction counter functions can't find package tcountcommon
![image](https://github.com/TPC-Council/HammerDB/assets/96612712/b6dd3f4a-8e39-4f00-924b-84889a0d7d37)

**Comment by sm-shaw** [comment]
The error is not related to the trace file conversion. 
Make sure you run HammerDB from the local directory i.e.

```
cd /home/hammerdb/HammerDB-4.10 
./hammerdb
```
This is related to this issue https://github.com/TPC-Council/HammerDB/issues/597. HammerDB is dependent 

Also note for replaying trace files - you must use the trace as a template to build your own workload. You should not expect a trace (if it modifies data) to run without errors.  You should make sure you run a test on the same data the test was taken from i.e use flashback or backup and restore/datapump. Nevertheless, if you have multiple sessions running at the same time you will need to synchronize the workloads correctly to make sure modify the data in the correct order. 

This is not something HammerDB does by default at present and there are no current plans to extend the trace file conversion.
